src/main/java/org/embulk/input/marketo/MarketoServiceImpl.java

```diff
@@ -0,0 +1,183 @@
+        }
+        catch (InterruptedException e) {
+            LOGGER.error("Exception when waiting for export job id: {}", exportID, e);
+            throw new DataException("Error when wait for bulk extract");
+        }
+        InputStream leadBulkExtractResult = marketoRestClient.getLeadBulkExtractResult(exportID);
+        final File tempDir = Files.createTempDir();
+        TempFileSpace tempFiles = new TempFileSpace(tempDir);
+        File tempFile = tempFiles.createTempFile(DEFAULT_FILE_FORMAT);
+        try {
+            Files.write(ByteStreams.toByteArray(leadBulkExtractResult), tempFile);
+        }
+        catch (IOException e) {
+            LOGGER.error("Encounter exception when download bulk extract file, job id [{}]", exportID, e);
+            throw new DataException("Can't download bulk extract file");
+        }
+        return tempFile;
+    }
+
+    @Override
+    public File extractAllActivity(Date startTime, Date endTime, int pollingTimeIntervalSecond, int bulkJobTimeoutSecond)
+    {
+        String exportID = marketoRestClient.createActitvityExtract(startTime, endTime, null);
+        marketoRestClient.startActitvityBulkExtract(exportID);
+        try {
+            marketoRestClient.waitActitvityExportJobComplete(exportID, pollingTimeIntervalSecond, bulkJobTimeoutSecond);
+        }
+        catch (InterruptedException e) {
+            LOGGER.error("Exception when waiting for export job id: {}", exportID, e);
+            throw new DataException("Error when wait for bulk extract");
+        }
+        InputStream activitiesBulkExtractResult = marketoRestClient.getActivitiesBulkExtractResult(exportID);
+        final File tempDir = Files.createTempDir();
+        TempFileSpace tempFiles = new TempFileSpace(tempDir);
+        File tempFile = tempFiles.createTempFile(DEFAULT_FILE_FORMAT);
+        try {
+            Files.write(ByteStreams.toByteArray(activitiesBulkExtractResult), tempFile);
+        }
+        catch (IOException e) {
+            LOGGER.error("Encounter exception when download bulk extract file, job id [{}]", exportID, e);
+            throw new DataException("Can't download bulk extract file");
+        }
+        return tempFile;
+    }
+
+    @Override
+    public Iterable<ObjectNode> getAllListLead(List<String> fieldNames)
+    {
+        RecordPagingIterable<ObjectNode> lists = marketoRestClient.getLists();
+        List<Iterable<ObjectNode>> iterables = new ArrayList<>();
+        for (ObjectNode node : lists) {
+            final String id = node.get("id").asText();
+            iterables.add(Iterables.transform(marketoRestClient.getLeadsByList(id, FluentIterable.from(fieldNames).filter(new Predicate<String>()
+            {
+                @Override
+                public boolean apply(@Nullable String input)
+                {
+                    return !LIST_ID_COLUMN_NAME.equals(input);
+                }
+            }).toList()), new Function<ObjectNode, ObjectNode>()
```

Comment:
This Function can be refactored as method as well.

Reply:
Fixed